modules/ROOT/pages/basketball-robot-guide/software/using-ide-plugins.adoc

```diff
@@ -0,0 +1,164 @@
+= Using IDE plugins
+
+There are plugins you can utilize in your development process to increase efficient.
+This document highlights some of them and provides examples of their usefulness.
+The IDE this document is based on is PyCharm but similar plugins are available for most other popular IDE-s.
+
+== SonarLint
+
+SonarLint is a static code analyzed that helps you write more maintainable code.
+It highlights issues, small and large, sometimes even suggesting fixes for them.
+It is a great tool to avoid the usual pitfalls when writing code.
+You can read more details about it here https://www.sonarsource.com/products/sonarlint/.
+
+=== Examining issues in a file
+
+SonarLint Scans your open file and highlights issues it discovered.
+In the plugin drawer you can see details about issues in the file.
+They are ordered based on severity.
+When selecting an issue you see additional information about siad issues on the right.
+There the basic nature of the issue is described and fixes are suggested.
+
+image::plugins_sonarlint_ide.png[]
+
+=== Example: Default values on mutable function parameters
+
+This is an usual issue where python assigns function default values once.
+If this value is modified, then so is the default used everywhere else.
+
+SonarLint provides a handy tip that describes the nature of the issue and sometimes has suggestions as to how to fix it.
+An example can be seen below when hovering over the issue highlighted in yellow.
+
+image::plugins_sonarlint_fix_suggestion.png[]
+
+Additional information can be seen in the plugin drawer below.
+It also highlights other issues that are in the file you are working on.
+We can see quite a few highlighted issues:
+
+image::plugins_sonarlint_file_info.png[]
+
+When clicking on the issue, you can see more details about the issue on the right.
+It can give you hints as to how to resolve the issue:
+
+image::plugins_sonarlint_aditional_info.png[]
+
+Some issues can be resolved automatically by clicking on the SonarLint suggested fix.
+In this case it suggests us to initialize the parameter.
+Clicking on it modifies your code.
+
+image::plugins_sonarlint_fix_suggestion.png[]
+
+The plugin modifies your code, resulting in new code, like so:
+
+image::plugins_sonarlint_fix_implementation.png[]
+
+== Mypy
+
+Mypy is an static type checker for Python.
+It extends python functionality to combine dynamic and static typing.
+This allows to avoid issues with mixed types where code logic can be affected by wrong but non-erroneous input types, like passing a float to a function requiring an integer.
```

Comment:
Split long sentences into multiple lines at the comma or some other logical place.